chemprop/cli/train.py

```diff
@@ -865,7 +866,21 @@ def train_model(
         trainer.fit(model, train_loader, val_loader)
 
         if test_loader is not None:
-            predss = trainer.predict(dataloaders=test_loader)
+            if isinstance(trainer.strategy, DDPStrategy):
+                torch.distributed.destroy_process_group()
+
+                best_ckpt_path = trainer.checkpoint_callback.best_model_path
+                trainer = pl.Trainer(
+                    logger=trainer_logger,
+                    enable_progress_bar=True,
+                    accelerator=args.accelerator,
+                    devices=1,
+                )
+                model = build_model(args, train_loader.dataset, output_transform, input_transforms)
+                model = model.load_from_checkpoint(best_ckpt_path)
```

Comment:
```suggestion
                model = model.load_from_checkpoint(best_ckpt_path)
```
I don't think you need to use `build_model` again. `load_from_checkpoint` is a class method so you could even do `MPNN.load_from_checkpoint()` without a model (though then you'd have to check if it is multicomponent). If you think `model = model.load_from_checkpoint(best_ckpt_path)` is unclear, you could consider `model = model.__class__.load_from_checkpoint(best_ckpt_path)`.

Reply:
Good catch! I noticed that we don't actually need to reload the model weights. The model for different processes in the DDP should have the same model weights, so we can just use it.

Reply:
I think you do need to reload the model weights because we want to use the best model, not the last one for testing. When we do `predss = trainer.predict(dataloaders=test_loader)` the trainer remembers which model checkpoint is the best and uses that but if we make a new trainer I don't think it has that information and will use the most recent model weights.

Reply:
@KnathanM You are right! Just change it back.